fabtests/common/shared.c

```diff
@@ -1247,33 +1247,80 @@ int ft_exchange_keys(struct fi_rma_iov *peer_iov)
 	return ret;
 }
 
-static void ft_close_fids(void)
-{
-	if (mr != &no_mr)
-		FT_CLOSE_FID(mr);
-	FT_CLOSE_FID(mc);
-	FT_CLOSE_FID(alias_ep);
-	FT_CLOSE_FID(ep);
-	FT_CLOSE_FID(pep);
-	FT_CLOSE_FID(pollset);
+int ft_close_fid(struct fid *fid)
+{
+	int ret = 0;
+	if (fid) {
+		ret = fi_close(fid);
+		if (ret)
+			FT_ERR("fi_close: %s(%d) fid %d", fi_strerror(-ret), ret, (int) fid->fclass);
+		fid = NULL;
```

Comment:
To work as a replacement for the FT_CLOSE_FID macro, this function needs to be:
```
int ft_close_fid(struct fid **fid)
```
Or setting fid = NULL does not propagate back to the caller.